OpenRA.Game/Map/Map.cs

```diff
@@ -804,6 +804,17 @@ public CPos ChooseRandomCell(MersenneTwister rand)
 			return new MPos(x, y).ToCPos(this);
 		}
 
+		public CPos ChooseClosestEdgeCell(CPos pos)
+		{
+			var x = pos.X < (Bounds.Right / 2) ? 0 : Bounds.Right;
+			var y = pos.Y < (Bounds.Bottom / 2) ? 0 : Bounds.Bottom;
```

Comment:
These 0s should be `Bounds.Left` and `Bounds.Top`, respectively. Or you could replace `Bounds.Right` and `Bounds.Bottom` with `MapSize.X` and `MapSize.Y`, but that would only yield fair results if the cordon had the same width on all sides.